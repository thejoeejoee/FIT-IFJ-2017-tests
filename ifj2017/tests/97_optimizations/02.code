' constant propagation into loop

scope
	dim counter as integer = 50
	dim constant as integer = 8 + 7

	do while counter > 0
		print (10.2 / 98 + (constant \ 3 * 6.85) - 14) / (constant - 5) + (constant + 9 - 8) * ((constant+17.21)\(10.8 - constant));
		counter = counter - 1
		print !"\n";
	loop

end scope