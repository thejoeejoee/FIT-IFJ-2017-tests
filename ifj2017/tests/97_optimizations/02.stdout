-129.39
-129.39
-129.39
-129.39
-129.39
-129.39
-129.39
-129.39
-129.39
-129.39
-129.39
-129.39
-129.39
-129.39
-129.39
-129.39
-129.39
-129.39
-129.39
-129.39
-129.39
-129.39
-129.39
-129.39
-129.39
-129.39
-129.39
-129.39
-129.39
-129.39
-129.39
-129.39
-129.39
-129.39
-129.39
-129.39
-129.39
-129.39
-129.39
-129.39
-129.39
-129.39
-129.39
-129.39
-129.39
-129.39
-129.39
-129.39
-129.39
-129.39
